transport/transport.go

```diff
@@ -185,13 +185,17 @@ type Stream struct {
 
 	headerChan chan struct{} // closed to indicate the end of header metadata.
 	headerDone uint32        // set when headerChan is closed. Used to avoid closing headerChan multiple times.
-	header     metadata.MD   // the received header metadata.
-	trailer    metadata.MD   // the key-value map of trailer metadata.
+	// hdrMu protects header and trailer metadata on the server-side.
+	hdrMu   sync.Mutex
+	header  metadata.MD // the received header metadata.
+	trailer metadata.MD // the key-value map of trailer metadata.
 
-	headerOk bool // becomes true from the first header is about to send
-	state    streamState
+	// On the server-side, headerOK is atomically set to 1 when the headers are sent out.
+	headerSent uint32
+	state      streamState
 
-	status *status.Status // the status error received from the server
+	// On client-side it is the status error received from the server.
```

Comment:
"Unused on server-side"?

Reply:
Yeah, it is unused. Not sure if you're suggesting that I add it or merely asking?

Reply:
I was asking, and now requesting you to add a comment like that since you confirmed that it's true.